### PR TITLE
add mir-opt section for optimization fuel

### LINKED ISSUE #962: Document optimization fuel for MIR optimizations
[This](https://github.com/rust-lang/rust/issues/77402#issuecomment-702248513) comment suggests adding some docs for optimization fuel usage. I agree with that sentiment, so now that https://github.com/rust-lang/rust/pull/79117 is merged I figure we can whip up a quick explanation for the change and why optimization fuels is used.

I'll write these docs, just wanted to make this issue in case it's needed for tracking purposes.

### PR DESCRIPTION
**Pull request by cjkenn**
Fixes #962

### CODE REVIEW

**Comment by cjkenn** [comment]
I'm not too familiar with how reviews are done for the book, @oli-obk would you like to take a look at this?

File: src/mir/optimizations.md [hunk 1]
```diff
@@ -60,3 +60,24 @@ want to query whether your optimization pass should run, you can check the
 current level using `tcx.sess.opts.debugging_opts.mir_opt_level`.
 
 [compiler MCP]: https://github.com/rust-lang/compiler-team/issues/319
+
+## Optimization fuel
+
+Optimization fuel is a compiler option (`-Z fuel=<crate>=<value>`) that allows for fine grained
+control over which optimizations can be applied during compilation: each optimization reduces
+fuel by 1, and when fuel reaches 0 no more optimizations are applied. This can help with debugging
+and identifying problems with optimizations.
```
Comment (oli-obk) [inline src/mir/optimizations.md:69]: Can you also leave a comment about this giving the ability to bisect optimizations by using a kind of binary search through the optimizations by using different fuels to home in on the problematic optimization? This explanation was the eye opener for me. I had no clue what fuel was for before your PR!

**Review by oli-obk — COMMENTED** [review]
Thanks! I'm also not sure about the review process around the guide, but pinging someone who reviewed or worked on the code related to the guide section is always good I think.

File: src/mir/optimizations.md (hunk 1 above)
Comment (cjkenn) [reply src/mir/optimizations.md]: I've added a small bit about that here. I hope that using terms like bisect and binary search make it clear how fuel can be used in this way...

**Review by oli-obk — APPROVED** [review]

File: src/mir/optimizations.md (hunk 1 above)
Comment (oli-obk) [reply src/mir/optimizations.md]: Thanks!

**Comment by oli-obk** [comment]
Oh I have merge powers here :laughing:

**Comment by oli-obk** [comment]
and I was too slow

**Comment by jyn514** [comment]
https://www.youtube.com/watch?v=LfmrHTdXgK4

**Pull request merged**